src/components/resource-picker/storage-location-screens.js

```diff
@@ -26,7 +26,19 @@ const SCREEN = {
   filesUpload: 'files-upload'
 };
 
-function StorageLocationScreens({ storageLocation, initialUrl, onSelect, onCancel }) {
+const updateFileDisplayName = (cdnObject, documentsTitles, t) => {
+  if (cdnObject.type === CDN_OBJECT_TYPE.directory) {
+    const documentTitle = documentsTitles.find(doc => doc._id === cdnObject.displayName);
+    cdnObject.displayName = `${documentTitle?.title || t('common:unknown')} [${cdnObject.displayName}]`;
```

Comment:
This is called every time `t` changes (e.g. the user switching the display language). But the `displayName` property is already patched after the first call, so the condition in line 31 will never be true again). This is a theoretical problem only, as currently the file dialog has to be closed in order to switch the display language, and after reopening it  there will be another call with a freshly fetched file list. This also plays a little bit into the general feeling I have about doing the human-readable-name patching on the client side (see also my - yet to be written - comment at the end of the PR).